File: src/main/java/com/kgaft/VulkanLib/Utils/DestroyableObject.java
package com.kgaft.VulkanLib.Utils;

public abstract class DestroyableObject {
    protected boolean destroyed = false;

    public synchronized void destroy(){
        this.destroyed = true;
    }


    @Override
    protected synchronized void finalize() throws Throwable {

        if(!destroyed){
            destroy();
        }

        super.finalize();
    }



}

File: VulkanLib/ObjectManagement/DestroyableObject.cs
namespace VulkanLib.ObjectManagement;

public abstract class DestroyableObject
{
    protected bool destroyed = false;

    public virtual void destroy()
    {
        this.destroyed = true;
    }

    ~DestroyableObject()
    {
        if (!destroyed)
        {
            destroy();
        }
    }
}